### PR TITLE
Reader interface

### PR DESCRIPTION
**Pull request by bradflood**
implement reader interface in anticipation of supporting both Stage B (current, LPTS Reading of xml file) and Stage C (querying of LANGUAGE and BIBLEBRAIN databases)

### CODE REVIEW

File: load/UpdateDBPAccessTable.py [hunk 1]
```diff
@@ -98,10 +98,10 @@ def _isPublicDomain(self, record):
 
 if (__name__ == '__main__'):
 	config = Config()
-	lptsReader = LPTSExtractReader(config.filename_lpts_xml)
+	languageReader = LanguageReaderCreator().create(config)
```
Comment (vichugofsl) [inline load/UpdateDBPAccessTable.py:101]: We should import the LanguageReaderCreator in the main to execute the UpdateDBPAccessTable.py file locally. e.g.

```python
from LanguageReaderCreator import LanguageReaderCreator
```

File: load/Validate.py [hunk 2]
```diff
@@ -112,23 +112,23 @@ def validateLPTS(self, filesets):
...
-				numeralsName = inp.lptsRecord.Numerals()
+				numeralsName = inp.languageRecord.Numerals()
 				if numeralsName != None and numeralsName not in self.numeralsSet:
 					logger.invalidValues(inp.stockNum(), "Numerals", numeralsName)
 
 
 if (__name__ == '__main__'):
 	from DBPLoadController import *
-	from LPTSExtractReader import *
+	from LanguageReader import *
```
Comment (vichugofsl) [inline load/Validate.py:126]: We should import the LanguageReaderCreator in the main to execute the Validate.py file locally. e.g.

```python
from LanguageReaderCreator import LanguageReaderCreator
```

File: load/UpdateDBPVideoTables.py [hunk 3]
```diff
@@ -184,15 +184,15 @@ def computeDurations(self, hashId):
 
 
 if (__name__ == '__main__'):
-	from LPTSExtractReader import *
+	from LanguageReader import *
 	from InputFileset import *
 	from DBPLoadController import *
```
Comment (vichugofsl) [inline load/UpdateDBPVideoTables.py:189]: we should import the LanguageReaderCreator in the main to execute the UpdateDBPVideoTables.py file locally. e.g.

```python
from LanguageReaderCreator import LanguageReaderCreator
```

File: load/UpdateDBPFilesetTables.py [hunk 4]
```diff
@@ -294,10 +294,10 @@ def getDuration(self, file):
 	from DBPLoadController import *
 
 	config = Config.shared()
-	lptsReader = LPTSExtractReader(config.filename_lpts_xml)
-	filesets = InputFileset.filesetCommandLineParser(config, lptsReader)
+	languageReader = LanguageReaderCreator().create(config)
+	filesets = InputFileset.filesetCommandLineParser(config, languageReader)
```
Comment (vichugofsl) [inline load/UpdateDBPFilesetTables.py:298]: @bradflood , I question by curious. I was checking the class InputFileset and I didn't see the method: filesetCommandLineParser. I tried to run locally this file I got the next error message:


```shell
AttributeError: type object 'InputFileset' has no attribute 'filesetCommandLineParser'
```

File: load/UpdateDBPFilesetTables.py [hunk 5]
```diff
@@ -294,10 +294,10 @@ def getDuration(self, file):
 	from DBPLoadController import *
 
 	config = Config.shared()
-	lptsReader = LPTSExtractReader(config.filename_lpts_xml)
-	filesets = InputFileset.filesetCommandLineParser(config, lptsReader)
+	languageReader = LanguageReaderCreator().create(config)
```
Comment (vichugofsl) [inline load/UpdateDBPFilesetTables.py:297]: I think that we should import the LanguageReaderCreator in the main to execute the UpdateDBPFilesetTables.py file locally. e.g.

```python
from LanguageReaderCreator import LanguageReaderCreator
```

**Review by vichugofsl — CHANGES_REQUESTED** [review]
@bradflood I have created some comments and I think we should add some changes.

**Review by vichugofsl — APPROVED** [review]

File: load/UpdateDBPFilesetTables.py (hunk 4 above)
Comment (bradflood) [reply load/UpdateDBPFilesetTables.py]: I've updated it with this:
languageReader = LanguageReaderCreator().create(config)
	filesets = InputProcessor.commandLineProcessor(config, AWSSession.shared().s3Client, languageReader)

**Pull request merged**